tests/scripts/mbedtls_test.py

```diff
@@ -75,12 +75,10 @@ def __escaped_split(inp_str, split_char):
         :param split_char: Split character
         :return: List of splits
         """
+        split_colon_fn = lambda x: re.sub(r'\\' + split_char, split_char, x)
         if len(split_char) > 1:
             raise ValueError('Expected split character. Found string!')
-        out = re.sub(r'(\\.)|' + split_char,
-                     lambda m: m.group(1) or '\n', inp_str,
-                     len(inp_str)).split('\n')
-        out = [x for x in out if x]
```

Comment:
Are we sure that this line is not needed?

Reply:
Honestly, I don't know. I am not a python expert, and I can't say why it was put here.
This line was accidentally removed... I'll check tomorow

Reply:
This line of code looks a bit suspicious to me, it is effectively removing all the items in the out list that evaluate to false. I think that the values in the out list can only be strings, so I think this would only remove empty strings from the out list. However, I dont see what situations would produce these empty strings...

Reply:
I don't know what could evaluate to none in this case, but I'm pretty sure it's not the empty strings, as out test data contains empty strings.
I have tried the following:
~~~
>>> inp_str1='exp:2:char*:""'
>>> print inp_str1
exp:2:char*:""
>>> out=map(lambda x: re.sub(r'\\'+split_char, split_char,x), re.split(r'(?<!\\)'+split_char, inp_str1))
>>> print out
['exp', '2', 'char*', '""']
>>> out = [x for x in out if x]
>>> print out
['exp', '2', 'char*', '""']
~~~
As you can see, the empty strings are not removed. Perhaps it's just corner case checking. note that once removing a "none" variable, it is likely that the script will raise an error on odd argument count.

I will return this line, even though I am not sure what it does,

Reply:
> As you can see, the empty strings are not removed.

I do not think there are any empty strings in your example. The string `'""'` is not empty as it contains the " characters

Reply:
Nevermind, I tried the following:
~~~
>>> inp_str1='exp:2:char*:'
>>> out=map(lambda x: re.sub(r'\\'+split_char, split_char,x), re.split(r'(?<!\\)'+split_char, inp_str1))
>>> print out
['exp', '2', 'char*', '']
>>> out = [x for x in out if x]
>>> print out
['exp', '2', 'char*']
~~~

Reply:
ok, thanks for the info. I did not know that we allowed terminating the test line with a :

Reply:
we don't allow this, but one can make mistakes.
This line was a measure of checking precondition exists.
also, I tried the following:
~~~
>>> inp_str1='exp::2::char*:'
>>> out=map(lambda x: re.sub(r'\\'+split_char, split_char,x), re.split(r'(?<!\\)'+split_char, inp_str1))
>>> print out
['exp', '', '2', '', 'char*', '']
>>> out = [x for x in out if x]
>>> print out
['exp', '2', 'char*']
~~~
So it's not only a matter of terminating the data with `:` , it's checking the `:` separator is as expected.
I'll return the line